### PR TITLE
[IMP] fieldservice_account: Simplify module

### PR DESCRIPTION
**Pull request by brian10048**
- This change removes most of the invoicing logic from fieldservice_account and puts it into its own module

### CODE REVIEW

**Comment by hparfr** [comment]
@brian10048  please have a look to https://github.com/brian10048/field-service/pull/8

**Comment by brian10048** [comment]
@hparfr @max3903 this is ready for further review

**Review by max3903 — APPROVED** [review]

**Review by hparfr — APPROVED** [review]
Thanks

File: fieldservice_account_analytic/__init__.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Copyright (C) 2018 Open Source Integrators
```
Comment (bodedra) [inline fieldservice_account_analytic/__init__.py:1]:
```suggestion
# Copyright (C) 2019 Open Source Integrators
```

File: fieldservice_account_analytic/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# Copyright (C) 2018 Open Source Integrators
```
Comment (bodedra) [inline fieldservice_account_analytic/__manifest__.py:1]:
```suggestion
# Copyright (C) 2019 Open Source Integrators
```

File: fieldservice_account_analytic/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    'name': 'Field Service - Analytic Accounting',
+    'summary': """Track analytic accounts on Field Service locations
+                  and orders""",
+    'version': '12.0.2.0.1',
```
Comment (bodedra) [inline fieldservice_account_analytic/__manifest__.py:8]:
```suggestion
    'version': '12.0.1.0.0',
```

File: fieldservice_account_analytic/models/account_invoice_line.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# Copyright (C) 2018 - TODAY, Open Source Integrators
```
Comment (bodedra) [inline fieldservice_account_analytic/models/account_invoice_line.py:1]:
```suggestion
# Copyright (C) 2019 Open Source Integrators
```

File: fieldservice_account_analytic/models/fsm_order.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
+# Copyright (C) 2018 - TODAY, Open Source Integrators
```
Comment (bodedra) [inline fieldservice_account_analytic/models/fsm_order.py:1]:
```suggestion
# Copyright (C) 2019 Open Source Integrators
```

File: fieldservice_account_analytic/tests/test_fsm_account_wizard.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
+# Copyright 2019 Ecosoft Co., Ltd (http://ecosoft.co.th/)
```
Comment (bodedra) [inline fieldservice_account_analytic/tests/test_fsm_account_wizard.py:1]:
```suggestion
# Copyright (C) 2019 Open Source Integrators
```

File: fieldservice_isp_account/__init__.py [hunk 7]
```diff
@@ -0,0 +1,4 @@
+# Copyright (C) 2018 Open Source Integrators
```
Comment (bodedra) [inline fieldservice_isp_account/__init__.py:1]:
```suggestion
# Copyright (C) 2019 Open Source Integrators
```

File: fieldservice_isp_account/__manifest__.py [hunk 8]
```diff
@@ -0,0 +1,26 @@
+# Copyright (C) 2018 Open Source Integrators
```
Comment (bodedra) [inline fieldservice_isp_account/__manifest__.py:1]:
```suggestion
# Copyright (C) 2019 Open Source Integrators
```

File: fieldservice_isp_account/__manifest__.py [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    'name': 'Field Service - ISP Accounting',
+    'summary': """Invoice Field Service orders based on employee time
+                  or contractor costs""",
+    'version': '12.0.2.0.1',
```
Comment (bodedra) [inline fieldservice_isp_account/__manifest__.py:8]:
```suggestion
    'version': '12.0.1.0.0',
```

File: fieldservice_isp_account/models/account_invoice_line.py [hunk 10]
```diff
@@ -0,0 +1,34 @@
+# Copyright (C) 2018 - TODAY, Open Source Integrators
```
Comment (bodedra) [inline fieldservice_isp_account/models/account_invoice_line.py:1]:
```suggestion
# Copyright (C) 2019 Open Source Integrators
```

File: fieldservice_isp_account/models/account_invoice_line.py [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models, _
+from odoo.exceptions import ValidationError
+
+
+class AccountInvoiceLine(models.Model):
+    _inherit = "account.invoice.line"
+
+    @api.onchange('product_id', 'quantity')
```
Comment (bodedra) [inline fieldservice_isp_account/models/account_invoice_line.py:11]:
```suggestion
    @api.onchange('product_id', 'quantity', 'fsm_order_id')
```

File: fieldservice_isp_account/models/fsm_order.py [hunk 12]
```diff
@@ -0,0 +1,184 @@
+# Copyright (C) 2018 - TODAY, Open Source Integrators
```
Comment (bodedra) [inline fieldservice_isp_account/models/fsm_order.py:1]:
```suggestion
# Copyright (C) 2019 Open Source Integrators
```

File: fieldservice_isp_account/models/fsm_order.py [hunk 13]
```diff
@@ -0,0 +1,184 @@
...
+        user = self.env['res.users'].browse(self.env.uid)
+        for order in self:
+            if user.employee_ids:
+                order.employee = True
+
+    @api.depends('employee_timesheet_ids', 'contractor_cost_ids')
+    def _compute_total_cost(self):
+        super()._compute_total_cost()
+        for order in self:
+            order.total_cost = 0.0
```
Comment (bodedra) [inline fieldservice_isp_account/models/fsm_order.py:42]:
```suggestion
            total_cost = 0.0
```

File: fieldservice_isp_account/models/fsm_order.py [hunk 14]
```diff
@@ -0,0 +1,184 @@
...
+            rate = 0
+            for line in order.employee_timesheet_ids:
+                rate = line.employee_id.timesheet_cost
+                order.total_cost += line.unit_amount * rate
```
Comment (bodedra) [inline fieldservice_isp_account/models/fsm_order.py:46]:
```suggestion
                total_cost += line.unit_amount * rate
```

File: fieldservice_isp_account/models/fsm_order.py [hunk 15]
```diff
@@ -0,0 +1,184 @@
...
+            for cost in order.contractor_cost_ids:
+                order.total_cost += cost.price_unit * cost.quantity
```
Comment (bodedra) [inline fieldservice_isp_account/models/fsm_order.py:48]:
```suggestion
                total_cost += cost.price_unit * cost.quantity
            order.total_cost = total_cost
```

File: fieldservice_isp_account/models/fsm_order.py [hunk 16]
```diff
@@ -0,0 +1,184 @@
...
+                else:
+                    raise ValidationError(_("The worker assigned to this order"
+                                            " is not a supplier"))
+            if order.employee_timesheet_ids:
+                order.account_stage = 'confirmed'
+
+    def account_create_invoice(self):
+        jrnl = self.env['account.journal'].search([
+            ('company_id', '=', self.env.user.company_id.id),
+            ('type', '=', 'sale'),
```
Comment (bodedra) [inline fieldservice_isp_account/models/fsm_order.py:110]:
```suggestion
            ('type', '=', 'sale')],
```

File: fieldservice_isp_account/models/fsm_order.py [hunk 17]
```diff
@@ -0,0 +1,184 @@
...
+            ('active', '=', True)],
```
Comment (bodedra) [inline fieldservice_isp_account/models/fsm_order.py:111]:
```suggestion
```

File: fieldservice_isp_account/tests/__init__.py [hunk 18]
```diff
@@ -0,0 +1,4 @@
+# Copyright 2019 Ecosoft Co., Ltd (http://ecosoft.co.th/)
```
Comment (bodedra) [inline fieldservice_isp_account/tests/__init__.py:1]:
```suggestion
# Copyright (C) 2019 Open Source Integrators
```

File: fieldservice_isp_account/tests/test_isp_account.py [hunk 19]
```diff
@@ -0,0 +1,178 @@
+# Copyright 2019 Ecosoft Co., Ltd (http://ecosoft.co.th/)
```
Comment (bodedra) [inline fieldservice_isp_account/tests/test_isp_account.py:1]:
```suggestion
# Copyright (C) 2019 Open Source Integrators
```

**Review by bodedra — APPROVED** [review]
Code review. Added minor improvements.

**Comment by bodedra** [comment]
Verified with @brian10048 
Please ignore my above suggested changes.

**Pull request merged**